lookbook/docs/patterns/19-attribute-help-texts.md.erb

```diff
@@ -0,0 +1,188 @@
+cases, such as if you are using a separate Form model.
+
+```erb
+<%%=
+  primer_form_with(url: "/foo", model: ProjectFormModel.new(project:)) do
+    render ProjectForm.new(f)
+  end
+%>
+```
+
+### For Component authors
+
+If you create a new input method (by extending
+`Primer::Forms::Dsl::Input` and adding an accompanying method to
+`lib/primer/open_project/forms/dsl/input_methods.rb`), care should be
+taken to call `decorate_options`, which enables the automatic rendering
+of help texts:
+
+```ruby
+  def super_duper_text_field(**)
+    add_input SuperDuperTextFieldInput.new(builder:, form:, **decorate_options(**))
+  end
+```
+
+You should also add an accompanying spec to
+`spec/lib/primer/open_project/forms/dsl/input_methods_spec.rb` to verify your
+input method supports help text-related arguments.
+
+Under-the-hood, automatic display of help texts for fields relies on rendering
+an `OpenProject::Common::AttributeLabelComponent`. There may be times when you
+need to create a component that has Form-field like behavior or visuals, but
+that does not use the Primer Forms API (i.e. components that extend
+`ApplicationComponent`/`ViewComponent::Base` rather than `Primer:Forms::Dsl::Input`).
+In such cases, you can use `AttributeLabelComponent` directly:
+
+```erb
+<%%=
+  flex_layout(align_items: :center) do |flex|
+    flex.with_column do
+      render(Primer::Beta::Text.new(font_weight: bold)) { project.name }
+    end
+
+    flex.with_column(mr: 2) do
+      render(
+        OpenProject::Common::AttributeLabelComponent.new(
+          attribute: :name,
+          model: project,
+          current_user: user
+        )
+      )
+    end
+  end
+%>
+```
+
+By default the label is wrapped in a `span`. This can overriden by passing the
+`tag:` argument. Passing `tag: nil` will remove the wrapper.
+
+## Attribute help texts outside of forms
+
+### View components
+
+You may also wish to show help texts when displaying attributes to the user. You
+can use `AttributeLabelComponent` as shown above, passing the model and
+applicable attribute. This takes care of finding an instantiating the Attribute
+help text.
+
+There is also a lower-level component available: `AttributeHelpTextComponent`.
+This component only renders an icon and optional additional text (via the
+`additional_label` slot). It renders without a wrapper and the caller is
+responsible for finding and passing an instance of `AttributeHelpText`.
+
+```erb
+<%%=
+  render OpenProject::Common::AttributeHelpTextComponent.new(help_text: @help_text) do |help_text|
+    help_text.with_additional_label { "Show Attribute help text" }
+  end
+%>
+```
+
+### Angular
+
+There is also an Angular `attribute-help-text` component available. It renders
+identically to its Rails `AttributeHelpTextComponent` counterpart. Set the
+attribute name using the `attribute` property and model name using
+`attributeScope` (in `PascalCase`).
```

Comment:
I think, we should mark that part somehow as deprected otherwise people will use again..